doc/api/esm.md

```diff
@@ -681,126 +695,54 @@ of the following:
 | `'wasm'`     | Load a WebAssembly module      | { [`ArrayBuffer`][], [`TypedArray`][] }                                    |
 
+Note: When format is `'commonjs'`, the value of `source` is ignored.
+
 Note: These types all correspond to classes defined in ECMAScript.
 
 * The specific [`ArrayBuffer`][] object is a [`SharedArrayBuffer`][].
 * The specific [`TypedArray`][] object is a [`Uint8Array`][].
 
-Note: If the source value of a text-based format (i.e., `'json'`, `'module'`) is
-not a string, it is converted to a string using [`util.TextDecoder`][].
-
-```js
-/**
- * @param {string} url
- * @param {Object} context (currently empty)
- * @param {Function} defaultGetFormat
- * @returns {Promise<{ format: string }>}
- */
-export async function getFormat(url, context, defaultGetFormat) {
-  if (Math.random() > 0.5) { // Some condition.
-    // For some or all URLs, do some custom logic for determining format.
-    // Always return an object of the form {format: <string>}, where the
-    // format is one of the strings in the preceding table.
-    return {
-      format: 'module',
-    };
-  }
-  // Defer to Node.js for all other URLs.
-  return defaultGetFormat(url, context, defaultGetFormat);
-}
-```
-
-#### `getSource(url, context, defaultGetSource)`
-
-> Note: The loaders API is being redesigned. This hook may disappear or its
-> signature may change. Do not rely on the API described below.
-
-* `url` {string}
-* `context` {Object}
-  * `format` {string}
-* `defaultGetSource` {Function}
-* Returns: {Object}
-  * `source` {string|SharedArrayBuffer|Uint8Array}
+If the source value of a text-based format (i.e., `'json'`, `'module'`)
+is not a string, it is converted to a string using [`util.TextDecoder`][].
 
-The `getSource` hook provides a way to define a custom method for retrieving
-the source code of an ES module specifier. This would allow a loader to
-potentially avoid reading files from disk.
+This hook can be used to simply map an unrecognised format to a supported
+format (ex `json6` to `module`) to allow Node.js to process it in a way that it
+already supports:
 
 ```js
 /**
  * @param {string} url
- * @param {{ format: string }} context
- * @param {Function} defaultGetSource
- * @returns {Promise<{ source: !(string | SharedArrayBuffer | Uint8Array) }>}
+ * @param {} context Currently not used
+ * @param {Function} defaultLoad
+ * @returns {Promise<{
+    format: !string,
+    source: !(string | SharedArrayBuffer | Uint8Array),
+  }>}
  */
-export async function getSource(url, context, defaultGetSource) {
-  const { format } = context;
-  if (Math.random() > 0.5) { // Some condition.
-    // For some or all URLs, do some custom logic for retrieving the source.
-    // Always return an object of the form {source: <string|buffer>}.
-    return {
-      source: '...',
-    };
+export async function load(url, context, defaultLoad) {
+  if (path.extname(url) === 'json6') {
```

Comment:
Again, I find `json6` confusing as it’s so close to `json`. I also prefer something like the previous examples, as they don’t imply any particular use case. We have realistic use cases farther down in the detailed examples (HTTPS loader, CoffeeScript loader).

```js
export async function load(url, context, defaultLoad) {
  const { format } = context;
  if (Math.random() > 0.5) { // Some condition.
    // For some or all URLs, do some custom logic for retrieving the source.
    // Always return an object of the form {format: <string>, source: <string|buffer>}.
    return {
      format,
      source: '...',
    }
  }
  // Defer to Node.js for all other URLs.
  return defaultLoad(url, context, defaultLoad);
}
```

Reply:
Hmm, I find these sorts of contrived, (IMHO, overly abstract) examples a waste of space—as a user, I wanna see how something might _actually_ be used (preferably with simple, medium, and advanced examples). A dev almost surely knows how `if` statements and `return`s work, so, such a generic sample provides nothing that the function doc above doesn't already cover.

If it's a choice between the generic example or nothing here, since there are the https and coffeescript examples below, I would pick nothing and just delete this.

Reply:
These “examples” are extremely minimal as they’re meant just to show what the function signatures are. The real examples are the CoffeeScript/HTTPS ones farther down.

Can we keep these for now and if you want to propose replacing `if (Math.random() > 0.5)` etc. with something else, that can be its own PR?